### PR TITLE
DEVDOCS-4711: [update] Handlebars reference, add strReplace helper

### PR DESCRIPTION
**Pull request by slsriehl**
# [DEVDOCS-4711]

## What changed?
* Refine external PR that adds previously-undocumented strReplace helper

## Anything else?
* #1723 

[DEVDOCS-4711]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-4711?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md [hunk 1]
```diff
@@ -46,6 +46,7 @@ The following table contains BigCommerce's open source [Handlebars helpers](http
 | [replace](#replace) | string | Replaces all instances of the first parameter in the second parameter. |
 | [setURLQueryParam](#seturlqueryparam) | string | Appends keys values to a URL. |
 | [stripQuerystring](#stripquerystring) | string | Removes a query string. |
+| [strReplace](#strreplace) | string | Replaces some or all occurrences of substring within the given string. |
```
Comment (bc-andreadao) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:49]:
```suggestion
| [strReplace](#strreplace) | string | Replaces some or all occurrences of a substring within the given string. |

```

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md [hunk 2]
```diff
@@ -1226,6 +1227,39 @@ Strips query string from a URL.
 
 - [See it in GitHub](https://github.com/bigcommerce/paper-handlebars/blob/master/helpers/stripQuerystring.js)
 
+### {{strReplace}}
+
+```handlebars
+{{strReplace string substr newSubstr occurrence}}
+```
+
+Replaces some or all occurrences of substring within the given string.
```
Comment (bc-andreadao) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1236]:
```suggestion
Replaces some or all occurrences of a substring within the given string.

```

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md [hunk 3]
```diff
@@ -1226,6 +1227,39 @@ Strips query string from a URL.
...
+
+```handlebars
+{{strReplace string substr newSubstr occurrence}}
+```
+
+Replaces some or all occurrences of substring within the given string.
+
+#### Parameters
+
+- `string` {String}: The original string.
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1240]:
```suggestion
- `subjectString` {String}: The original string to modify.

```

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md [hunk 4]
```diff
@@ -1226,6 +1227,39 @@ Strips query string from a URL.
 
 - [See it in GitHub](https://github.com/bigcommerce/paper-handlebars/blob/master/helpers/stripQuerystring.js)
 
+### {{strReplace}}
+
+```handlebars
+{{strReplace string substr newSubstr occurrence}}
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1233]:
```suggestion
{{strReplace subjectString targetSubstring newSubstring occurrenceSelection}}

```

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md (hunk 2 above)
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1236]: alternatively:

```suggestion
Replaces some or all occurrences of the target substring within the subject string.

```

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md [hunk 5]
```diff
@@ -1226,6 +1227,39 @@ Strips query string from a URL.
...
+```handlebars
+{{strReplace string substr newSubstr occurrence}}
+```
+
+Replaces some or all occurrences of substring within the given string.
+
+#### Parameters
+
+- `string` {String}: The original string.
+- `substr` {String}: The substring in `string` to be replaced.
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1241]:
```suggestion
- `targetSubstring` {String}: The substring in `string` to be replaced.

```

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md [hunk 6]
```diff
@@ -1226,6 +1227,39 @@ Strips query string from a URL.
...
+- `newSubstr` {String}: The substring to replace `substr`.
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1242]:
```suggestion
- `newSubstring` {String}: The substring you want to insert in place of `targetSubstring`.

```

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md [hunk 7]
```diff
@@ -1226,6 +1227,39 @@ Strips query string from a URL.
...
+- `occurence` {Number}: The number of occurrences of `substr` to be replaced by `newSubstr` in `string`.
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1243]:
```suggestion
- `occurrenceSelection` {Integer}: Optional. The number of occurrences of `targetSubstring` to replace with `newSubstring`. If you pass no argument or the supplied integer is greater than the number of `targetSubstring` occurrences, `newSubstring` will replace all instances of `targetSubstring`. A positive integer will replace `targetSubstring` starting from the beginning of the line;  a negative integer will replace from the end of the line.

```

Comment (slsriehl) [reply docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1243]: test end of line condition and modify this description or line 1251-1252 accordinly.

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md (hunk 1 above)
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:49]: alternatively:

```suggestion
| [strReplace](#strreplace) | string | Replaces some or all occurrences of a target substring within the subject string. |

```

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md (hunk 7 above)
Comment (slsriehl) [reply docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:1245]: suggestion updated following collaboration.

File: docs/stencil-docs/reference-docs/handlebars-helpers-reference.md [hunk 8]
```diff
@@ -46,6 +46,7 @@ The following table contains BigCommerce's open source [Handlebars helpers](http
 | [replace](#replace) | string | Replaces all instances of the first parameter in the second parameter. |
 | [setURLQueryParam](#seturlqueryparam) | string | Appends keys values to a URL. |
 | [stripQuerystring](#stripquerystring) | string | Removes a query string. |
+| [strReplace](#strreplace) | string | Replaces some or all occurrences of a target substring within the subject string. |
```
Comment (slsriehl) [inline docs/stencil-docs/reference-docs/handlebars-helpers-reference.md:49]:
```suggestion
| [strReplace](#strreplace) | string | Replaces some or all occurrences of a target substring within a subject string. |

```

**Comment by slsriehl** [comment]
@bc-traciporter your work on this was stellar. 🚀

**Pull request merged**